crates/sui-graphql-e2e-tests/tests/call/coin_metadata.move

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//# init --addresses test=0x0 --accounts A --simulator
+
+//# publish --sender A
+module test::fake {
+    use std::option;
+    use sui::coin;
+    use sui::transfer;
+    use sui::tx_context::{Self, TxContext};
+
+    struct FAKE has drop {}
+
+    fun init(witness: FAKE, ctx: &mut TxContext){
+        let (treasury_cap, metadata) = coin::create_currency(witness, 2, b"FAKE", b"", b"", option::none(), ctx);
+        transfer::public_freeze_object(metadata);
+        transfer::public_transfer(treasury_cap, tx_context::sender(ctx));
+    }
+}
+
+//# create-checkpoint
+
+//# run-graphql
+{
+  coinMetadata(coinType: "0x997887d141bd15d9e449d7f46da3eae759f6abfaf30e42d5f81cc26c185c85e3::fake::FAKE") {
```

Comment:
Yeah ...you can't hardcode package IDs here -- that way madness lies (for you and anyone who has to update this test, which includes anyone who touches the genesis objects, like the frameworks).

Unfortunately, I don't think the support for named addresses that @oxade added recently is quite enough because you only get a variable binding `test` to the address, and GraphQL has no way to interpolate strings, so we may have to operate outside of GraphQL and add some kind of string interpolation for these GraphQL queries?

Reply:
This is an interesting case. I'm open to adding support for this but I wanna come up with a holistic solution over a on-off hack. Are there other adjacent features we might want?
cc @wlmyng @stefan-mysten?

Reply:
My thinking here is that we can create something quite general by building on what you already have, and implementing the variable interpolation in the transactional test runner, rather than relying on GraphQL's built-in variable support.  E.g. you look for strings like `$obj(x, y)` or `$pkg(name)` and substitute those for the appropriate ID, such that Will's test becomes:

```graphql
{
  coinMetadata(coinType: "$pkg(test)::fake::FAKE") { ... }
}
```

Reply:
Got it. Gonna think more about this and other approaches. In meantime we can get this PR through for 1.5, and let me know of other features we might want to the testing system.